File: 0128-longest-consecutive-sequence/solution.java
class Solution {
    public int longestConsecutive(int[] nums) {
        int ans = 0;
        var hashSet = new HashSet<Integer>();

        for (int i = 0; i < nums.length; i++) {
            hashSet.add(nums[i]);
        }

        for (int i = 0; i < nums.length; i++) {
            if (!hashSet.contains(nums[i] - 1)) {
                int count = 0;
                int j = nums[i];
                while (hashSet.contains(j)) {
                    hashSet.remove(j);
                    count++;
                    j++;
                }
                ans = Math.max(ans, count);

            }
        }
        return ans;
    }
}

File: 0128-longest-consecutive-sequence/solution.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        nums_set = set(nums)
        best = 0

        for x in nums_set:
            if x - 1 not in nums_set:
                temp = x
                res = 1

                while temp + 1 in nums_set:
                    res+= 1
                    temp += 1
                
                best = max(best, res)
        
        return best
        
